common/Fabric-Install-Utilities.psm1

```diff
@@ -277,123 +277,170 @@ function Get-AccessToken($authUrl, $clientId, $scope, $secret)
     return $accessTokenResponse.access_token
 }
 
-function Add-ApiRegistration($authUrl, $body, $accessToken)
+function Save-ApiRegistration($authUrl, $body, $accessToken)
 {
     $url = "$authUrl/api/apiresource"
     $headers = @{"Accept" = "application/json"}
     if($accessToken){
         $headers.Add("Authorization", "Bearer $accessToken")
     }
-    $registrationResponse = Invoke-RestMethod -Method Post -Uri $url -Body $body -ContentType "application/json" -Headers $headers
-    return $registrationResponse.apiSecret    
+    
+    try{
+        $registrationResponse = Invoke-RestMethod -Method Post -Uri $url -Body $body -ContentType "application/json" -Headers $headers
+        return $registrationResponse.apiSecret
+    }catch{
+        $exception = $_.Exception
+        $apiResourceObject = ConvertFrom-Json -InputObject $body
+        if ($exception -ne $null -and $exception.Response.StatusCode.value__ -eq 409) {
+            Write-Success "API Resource $($apiResourceObject.name) is already registered...updating registration settings."
+            Write-Host ""
+            try{
+                $registrationResponse = Invoke-RestMethod -Method Put -Uri "$url/$($apiResourceObject.name)" -Body $body -ContentType "application/json" -Headers $headers
+                return ""
+            }catch{
+                $exception = $_.Exception
+                $error = Get-ErrorFromResponse -response $exception.Response
+                Write-Error "There was an error updating API resource $($apiResourceObject.name): $error. Halting installation."
+                throw $exception
+            }
+        }
+        else {
+            Write-Error "Could not register API $($apiResourceObject.name) with Fabric.Identity, halting installation."
+            throw $exception
+        }
+    }
 }
 
-function Add-ClientRegistration($authUrl, $body, $accessToken)
+function Save-ClientRegistration($authUrl, $body, $accessToken)
```

Comment:
Renaming this function is a breaking change for others who might be using it, Perhaps we keep the name the same?